File: crates/klyron_transpiler/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_transpiler {
TranspilerBuilder::TranspilerBuilder() : config_() {}
TranspilerBuilder& TranspilerBuilder::enabled(bool v) { config_.enabled = v; return *this; }
TranspilerBuilder& TranspilerBuilder::verbose(bool v) { config_.verbose = v; return *this; }
TranspilerConfig TranspilerBuilder::build() { return config_; }
}

File: crates/klyron_transpiler/bindings/rust/builder.rs
use crate::config::TranspilerConfig;

pub struct TranspilerBuilder {
    config: TranspilerConfig,
}

impl TranspilerBuilder {
    pub fn new() -> Self { Self { config: TranspilerConfig::default() } }
    pub fn build(self) -> TranspilerConfig { self.config }
}

impl Default for TranspilerBuilder {
    fn default() -> Self { Self::new() }
}
